lib/xdr/accounts/error.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Stellar.XDR.ThresholdsError do
+  @moduledoc """
+  Module to handle exceptions that may arise from the `Stellar.XDR.Thresholds` implementation.
+  """
+  @type t :: %__MODULE__{message: String.t()}
+
+  defexception [:message]
+
+  @spec exception(type :: atom()) :: no_return()
+  def exception(:invalid_thresholds_specification),
+    do: %__MODULE__{
+      message:
+        "Invalid thresholds specification. Thresholds must be provided as a keyword list -> master_weight: master_weight, low: low, med: med, high: high"
```

Comment:
@JuanCamiloPerilla I'm not sure if this message is still valid after the changes discussed.

Reply:
We can simplify this message and make it more specific to the current implementation. 

I propose adding the following statement:

```elixir
 message:
    "Function clause match error. An keyword list for the keys master_weight, low, med and high, respectively, was expected."
```

It is also worth mentioning that, in my opinion, the usage of this Thresholds data structure is really domain specific. People using it are most likely to know what they're doing. Do we really need error handling here? Or is the API intuitive enough by itself?

At least we can think of providing hints on what the actual order of the parameters should be.

Reply:
@JuanCamiloPerilla IMO, the error handling on the `encode_xdr/2` is not really needed since you're already validating the structs arguments when invoking `new/1`.